comid/digests.go

```diff
@@ -41,6 +43,54 @@ func (o Digests) Valid() error {
 	return nil
 }
 
+func (o *Digests) Equal(r *Digests) bool {
+	return reflect.DeepEqual(o, r)
```

Comment:
I don't think this is correct. The order of digests entries is semantically irrelevant, so I don't think to `Digests` instances that differ only in the order of their entries (but the entries, when reordered match).  

I think it should be something like this:
```go
func (o *Digests) Equal(r *Digests) bool {
	om := make(map[uint64][][]byte)
	for _, oe := range *o {
		vs, ok := om[oe.HashAlgID]
		if ok {
			om[oe.HashAlgID] = append(vs, oe.HashValue)
		} else {
			om[oe.HashAlgID] = [][]byte{oe.HashValue}
		}
	}

	outer:
	for _, re := range *r {
		ovs, ok := om[re.HashAlgID]
		if !ok {
			return false
		}

		for _, ov := range ovs {
			if bytes.Equal(ov, re.HashValue) {
				continue outer
			}
		}

		return false
	}

	return true
}
```

Reply:
An argument can also be made for simply using the `CompareAgainstReference` implementation. If, semantically, `Digests` contains different hashes of the same object , so as long as the overlap matches, the underlying objects are equal, thefore the `Digests` can be said to be equal.

Reply:
You’re correct, that the order of entries don’t matter. Let me fix that. Thanks for bringing it up.

I don’t think reusing `CompareAgainstReference`, which does a partial match, confirms to the Equality semantics we’re going for here.